rust_dev_preview/examples/auto-scaling/src/bin/scenario.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::BTreeSet;
+
+use anyhow::anyhow;
+use autoscaling_code_examples::scenario::AutoScalingScenario;
+use tracing::{info, warn};
+
+async fn show_scenario_description(scenario: &AutoScalingScenario, event: &str) {
+    let auto_scaling_scenario_description = scenario.describe_scenario().await;
+    match auto_scaling_scenario_description {
+        Ok(description) => info!("DescribeAutoScalingInstances: {event}\n{description}"),
+        Err(err) => info!("Error in DescribeAutoScalingInstances: {event}\n{err:?}"),
+    }
+}
+
+#[derive(Default, Debug)]
+struct Warnings(Vec<String>);
+
+impl Warnings {
+    pub fn push(&mut self, warning: &str, error: anyhow::Error) {
+        let formatted = format!("{warning}: {error:?}");
+        warn!("{formatted}");
+        self.0.push(formatted);
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+}
```

Comment:
just curious—why use this instead of tracing::warn?

Reply:
In the bin, it's a nice way to gather all the errors for one final write to stdout at the end of the run.